package/run.ps1

```diff
@@ -46,6 +46,40 @@ function Get-StringHash {
     return (Get-FileHash -InputStream $stringAsStream -Algorithm SHA256).Hash.ToLower()
 }
 
+# stops the RKE2 process as well as other processes spawned from rke2
+function Stop-RKE2-Processes () {
+    $rke2ProcessNames = @('rke2', 'kube-proxy', 'kubelet', 'containerd', 'calico-node')
+    foreach ($processName in $rke2ProcessNames) {
+        Write-LogInfo "Checking if $processName process exists"
+        if (Get-Process -Name $processName -ErrorAction SilentlyContinue) {
+            Write-LogInfo "$processName process found, stopping now"
+            Stop-Process -Name $processName
+        }
+    }
+
+    # ensure all processes have exited
+    $waitLimit = 3
+    foreach ($processName in $rke2ProcessNames) {
+        $currentWait = 0
+        while ($waitLimit -ne $currentWait) {
+            # check if the process still exists
+            Get-Process -Name $processName -ErrorAction SilentlyContinue -ErrorVariable processNotFoundError | out-null
+            if ($processNotFoundError) {
+                Write-LogInfo "$processName exited successfully"
+                break
+            }
+
+            # wait for a bit for the process to exit
+            Write-LogInfo "Waiting for $processName process to stop"
+            $currentWait++
+            if ($waitLimit -eq $currentWait) {
+                Write-LogFatal "Failed to stop all RKE2 processes: timed out waiting for $processName to exit."
```

Comment:
nit: Do we need to add a break here after the fatal? It won't get executed, but wonder if we should still add it in case we ever change this to a LogError for some reason (just for general clean looking code)

Reply:
yeah, it can't hurt.